app/controllers/miq_ae_class_controller.rb

```diff
@@ -2587,9 +2593,12 @@ def ns_right_cell_text
   end
 
   def ordered_domains_for_priority_edit_screen
-    MiqAeDomain.order('priority DESC')
+    Tenant.current_tenant.visible_domains.sort_by{ |d| d.priority }.reverse
```

Comment:
@h-kataria 
Can we just use the Tenant.current_tenant.editable_domains?
That would give you a sorted list of domains and will only include the domains that the user can edit.